src/orange/detail/BIHData.hh

```diff
@@ -24,37 +24,43 @@ namespace detail
 /*!
  * Data for a single inner node in a Bounding Interval Hierarchy.
  *
- * Note that the LEFT bounding plane position is the far right boundary of the
- * left side of the tree, and the RIGHT bounding plane position is the far left
- * boundary of the right side of the tree. Since the halfspaces of created by
- * the bounding planes may overlap, the LEFT bounding plane position could be
- * either left or right of the RIGHT bounding plane position.
+ * As a convention, a node's LEFT edge corresponds to the half space that is
+ * less than the partition value. In other words, the LEFT bounding plane
+ * position is the far right boundary of the left side of the tree, and the
+ * RIGHT bounding plane position is the far left boundary of the right side of
+ * the tree. Since the halfspaces created by the bounding planes may overlap,
+ * the LEFT bounding plane position could be either left or right of the RIGHT
+ * bounding plane position.
```

Comment:
Another (less ambiguous) way to think about this is that the *center* of the left/lower region is always less than the *center* of the right/upper half-space?

Reply:
I don't think the term "center" is ideal here since these are infinite half spaces. I think what I have is fairly explicit, and I think it's important to state it in these terms even if it is a bit wordy. In what way do you think it is ambiguous?

Reply:
They're not really infinite since they're the union of all the bounding boxes of all the daughter nodes, right? (All the infinite bboxes are excluded.) I wasn't suggesting you take out any of the text, which does make sense, but thought it could be another way of thinking of the tree layout.

Reply:
@sethrj there are two separate concepts here: bounding planes, and the edge bounding boxes. Both of which are found on every edge.

When traversing the tree for point-in-vol,  the decision to traverse an edge is only contingent on the bounding plane of the edge. Bounding planes always create infinite half spaces.

When traversing the tree for intersections, the decision to traverse an edge is only contingent on the pre-computed edge bounding boxes. Edge bounding boxes are not the union on the bounding boxes of the daughter nodes --- that would create something more like a BVH. Edge bounding boxes can be thought of as "what we know so far" as the tree is traversed. They are the intersection of half spaces created by all bounding planes between the root and the current edge. Many of the edge bounding boxes are infinite in a least one direction. In fact, the leftmost and rightmost edges at every depth, i.e. LL...L and RR...R, are always infinite on the + or - side of every direction, because the represent the case that is always on the + or - side of every bounding plane.